### ISSUE TITLE
Retry won't work with Kotlin parametrised tests enclosed in backticks

### ISSUE THREAD

**Issue by ivans083** [gradle/test-retry-gradle-plugin#123]
Sample with Kotlin and JUnit5:

```
    @MethodSource("anySource")
    @ParameterizedTest
    fun `test method with spaces`(from: Param, to: Param) {
       Assertions.assertTrue(false)
    }
```

Caused by: java.lang.IllegalStateException: org.gradle.test-retry was unable to retry the following test methods, which is unexpected. Please file a bug report at https://github.com/gradle/test-retry-gradle-plugin/issues
   TestsClass#test method with spaces(Param, Param)[1]
   TestsClass#test method with spaces(Param, Param)[6]

**Comment by PTParta** [comment]
I'm having the same issue. Has there been progress with this issue? Before I start to change the names of hundreds of test cases it would be nice to know if this has been fixed :)

**Comment by ivans083** [comment]
Hey Pauli, I've seen that issue too, there are 2 separate issues:
1. Backticks on @ParametrizedTest
2. Parentesis (and I believe maybe some other special chars too) on @Test.

I tested last time a month ago, still not resolved, had to go for renaming
test names and avoiding @ParametrizedTest

Cheers,

Ivan.-

**Comment by Sobolev-Ilya** [comment]
Is there any workarounds for case with parentheses in test method name?

Tests with this names wont be retried
```
@Test
    fun `This test would not retried()`()

@Test
    fun `This test would not retried (too)`()

```

But this tests will be retried (when paretheses not at the end)
```
@Test
    fun `This test (would) be retries`()
```

**Comment by jean-andre-gauthier** [comment]
This has been fixed in https://github.com/gradle/test-retry-gradle-plugin/pull/272, and will be released in `1.5.8`
